### PR TITLE
Document generating JSON schema

### PR DESCRIPTION
**Pull request by kciesielski**
A follow-up to https://github.com/softwaremill/tapir/pull/2873

### CODE REVIEW

File: doc/docs/json-schema.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  val jsonSchema: ReferenceOr[ASchema] = TapirSchemaToJsonSchema(
+    tSchema,
+    markOptionsAsNullable = true,
+    metaSchema = MetaSchemaDraft04, // default
+    schemaName = defaultSchemaName // default
+)
+```
+
+All the nested schemas will be referenced from the `$defs` element.
+In order to generate a JSON representation of the schema, you can use Circe:
```
Comment (adamw) [inline doc/docs/json-schema.md:36]: that would be a good place to mention the extra dependency that's needed

File: doc/docs/json-schema.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Generating JSON Schema
+
+Since Tapir v1.4.0 you can conveniently generate JSON schema from Tapir schema, which can be derived from your Scala types. Use `TapirSchemaToJsonSchema`:
```
Comment (adamw) [inline doc/docs/json-schema.md:3]: Let's skip the "since 1.4.0" - release notes play that role

**Review by adamw — APPROVED** [review]

File: doc/docs/json-schema.md (hunk 1 above)
Comment (kciesielski) [reply doc/docs/json-schema.md:36]: It's mentioned here https://github.com/softwaremill/tapir/pull/2904/files#diff-a2236d1f0584452eefe52cb190b04ffdf887407b7237a121fb838c3df9017325R8
Do you think we should repeat that here?

Comment (kciesielski) [reply doc/docs/json-schema.md]: Ok, I rearranged the document to divide more clearly between conversion and serialization, including imports. I also added some more fixes.

**Review by adamw — APPROVED** [review]

**Comment by adamw** [comment]
Looks good now, thanks :)

**Pull request merged**